gdrive

```diff
@@ -235,23 +253,28 @@ def main():
 
     service = init()
     if service is None:
-        print('Failed to start service. Exiting..')
+        print('Failed to start service. Exiting...')
         return
 
     opt = read_argv()
-    # If there is an ID, downloads with that ID
-    # If there is a name, downloads with that name
-    # If there is nothing, downloads with the third argument
     if func == 'download':
+        # Downloads with the given file ID
         if opt['file_id']:
             download(service, opt['file_id'])
+        # --last overrides --name if true
+        elif opt['filename'] or opt['download_last']:
```

Comment:
You're still doing the if inside of the "download_search" fucntion.

I would prefer to have something like the following:
`elif opt['filename']:
    download_search(service, filename=opt['filename'])
elif opt['download_last']:
    download_search(service, download_last=opt['download_last']
else:
 ...`

and in download_search function, you would have something like:

`def download_search(service, filename=None, download_last=False):
    files_found, _ = search_filename(service, filename, 1, download_last)
    if not files_found:
        ...`

Reply:
I thought this would be formatted, but apparently not. Check if you can understand what I'm saying.